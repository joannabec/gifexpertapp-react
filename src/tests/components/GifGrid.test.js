/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
// mock de la función para asignar manualmente valores de retorno
jest.mock('../../hooks/useFetchGifs');

describe('Testeando <GifGrid />', () => {
  const category = 'Avatar';

  test('Debería renderizar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería mostrarse la categoría como subtitulo', () => {
    const wrapper = shallow(<GifGrid category={category} />);
    const element = wrapper.find('h3');
    expect(element.text()).toBe('Avatar');
  });

  test('Debería mostrar items cuando se cargan imágenes de useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://cualquiercosa/imagen.jpg',
        title: 'Alguna descripción',
      },
      {
        id: '123',
        url: 'https://cualquiercosa/imagen.jpg',
        title: 'Alguna descripción',
      },
    ];
    // mockReturnValue para asignar los valores de retorno de la función
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
