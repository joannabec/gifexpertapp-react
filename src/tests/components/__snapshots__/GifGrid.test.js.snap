// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando <GifGrid /> Deber√≠a renderizar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Avatar
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-container"
  />
</Fragment>
`;
